Sources/PT.PM.Common/TextUtils.cs

```diff
@@ -382,5 +384,64 @@ private static void ParseLineColumn(ReadOnlySpan<char> text, out int line, out i
                 throw new FormatException($"Invalid or too big column value {value} while {nameof(LineColumnTextSpan)} parsing.");
             }
         }
+
+        public static Literal CreateNumericLiteral(string value, TextSpan textSpan, int fromBase = 10)
+        {
+            switch (fromBase)
+            {
+                case 2:
+                    try
+                    {
+                        return new LongLiteral(Convert.ToInt64(value, 2), textSpan);
+                    }
+                    catch
+                    {
+                        BigInteger bigIntValue = 0;
+                        foreach (var c in value)
+                        {
+                            bigIntValue <<= 1;
+                            bigIntValue += c == '1' ? 1 : 0;
+                        }
+                        return new BigIntLiteral(bigIntValue, textSpan);
+                    }
+                case 8:
+                    try
+                    {
+                        return new LongLiteral(Convert.ToInt64(value, 8), textSpan);
+                    }
+                    catch
+                    {
+                        return new BigIntLiteral(value.Aggregate(new BigInteger(),
+                            (bigInt, c) => bigInt * 8 + c - '0'));
+                    }
+                case 10:
+                    if (int.TryParse(value, out int intValue))
+                    {
+                        return new IntLiteral(intValue, textSpan);
+                    }
+                    else if (long.TryParse(value, out long longValue))
+                    {
+                        return new LongLiteral(longValue, textSpan);
+                    }
+                    else
+                    {
+                        return new BigIntLiteral(BigInteger.Parse(value), textSpan);
+                    }
+                case 16:
+                    value = value.StartsWith("0X", StringComparison.OrdinalIgnoreCase)
+                        ? value.Substring(2)
+                        : value;
+                    try
+                    {
+                        return new LongLiteral(Convert.ToInt64(value, 16), textSpan);
+                    }
+                    catch
+                    {
+                        return new BigIntLiteral(BigInteger.Parse(value, System.Globalization.NumberStyles.HexNumber));
+                    }
+                default:
+                    throw new NotSupportedException($"{fromBase} base сonversion is not supported");
+            }
+        }
```

Comment:
I suggest wrapping this method to `try catch` block and return default `IntLiteral` with value of `0` if exception. Conversion step should be reliable as possible.